orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/config/WebhookConfiguration.java

```diff
@@ -124,6 +138,127 @@ public ClientHttpRequestFactory webhookRequestFactory(
     return requestFactory;
   }
 
+  /** Return if the request size is valid. Throw an exception otherwise. */
+  private void validateRequestSize(Request request, long maxRequestBytes) throws IOException {
+    if (maxRequestBytes <= 0L) {
+      return;
+    }
+
+    long headerSize = request.headers().byteCount();
+
+    // If the headers are already too big, no need to deal with the body size
+    if (headerSize > maxRequestBytes) {
+      String message =
+          String.format(
+              "rejecting request to %s with %s byte(s) of headers, maxRequestBytes: %s",
+              request.url(), headerSize, maxRequestBytes);
+      log.info(message);
+      throw new IllegalArgumentException(message);
+    }
+
+    RequestBody requestBody = request.body();
+    if (requestBody == null) {
+      return;
+    }
+
+    long contentLength = 0L;
+    try {
+      contentLength = requestBody.contentLength();
+      if (contentLength == -1) {
+        // Given that OkHttp3ClientHttpRequestFactory.buildRequest (called by
```

Comment:
I believe this is possible. I think this occurs for streaming, e.g. trying to send a request with a massive file returns content length == -1. See chunked encoding.

Reply:
Now, if webhooks can't do chunked encoding, then we can assume `-1` wont be the value.

Reply:
I'll admit I haven't tried that hard to send a streamed request via a webhook stage.  I don't *think* it's possible though since the request is defined in the pipeline config.  Of course with SpEL expressions, etc. it can be dynamic and huge, but I still think it ends up being a String with a known length by the time we get here.